### ISSUE TITLE
Attributes not working in managed converters

### ISSUE THREAD

**Issue by reiudo** [omnifaces/omnifaces#437]
Consider the following page snipped and converter implementation:
```
<h:form id="TestPage">
    <h:commandButton id="button" type="submit" value="Button 1" />
    <br/>
    <h:inputText id="someText" value="#{bean.someValue}" >
        <o:converter converterId="com.pass.project.jsf23test.TestConverter" foo="foo value" />
    </h:inputText>
    
</h:form>
```

```
@FacesConverter(value=TestConverter.NAME, managed=true)
public class TestConverter implements Converter<String> {

    public static final String NAME = "com.pass.project.jsf23test.TestConverter";

    @Inject @Any
    private Bean bean;
    
    private String foo;
    
    @Override
    public String getAsObject(FacesContext context, UIComponent component, String value) {
        System.out.println("getAsObject - foo: " + foo + " bean: " + bean);
        return value;
    }

    @Override
    public String getAsString(FacesContext context, UIComponent component, String value) {
        return value;
    }
    
    public String getFoo() {
        return foo;
    }

    public void setFoo(String foo) {
        System.out.println("TestConverter.setFoo(foo): " + foo);
        this.foo = foo;
    }
}
```

With `managed=false`, foo will be populated correctly with `foo value`. Unfortunately ‘bean’ will be null then.

With `managed=true` foo won’t be populated (but bean has a value of course).

Environment: Omnifaces 3.0.0 and Mojarra 2.3.3.

**Comment by reiudo** [comment]
I've added a minimal project to reproduce the issue.

[Jsf23TestWithOmnifaces3.zip](https://github.com/omnifaces/omnifaces/files/1739164/Jsf23TestWithOmnifaces3.zip)

**Comment by reiudo** [comment]
Without `@FacesConfig(version=Version.JSF_2_3)` the attributes are working. But that is not an option.

**Comment by BalusC** [comment]
Fix is available in today's latest 3.1-SNAPSHOT.

Workaround for OmniFaces 3.0: remove parameterized type from `Converter` (so it becomes raw), remove `managed=true` from annotation, use `bean-discovery-mode="all"` in `beans.xml` or put `@Dependent` on converter, then OmniFaces will continue to manage it.

From OmniFaces 3.1 on, parameterized `Converter<T>` is now also for OmniFaces-managed CDI injection (as it was in OmniFaces 2.x), but `<o:converter>` won't support JSF's `managed=true` for now. Internally, Mojarra wraps the converter instance in a decorating instance which doesn't at all have the necessary setter methods and thus the attributes are silently ignored. I have clarified `<o:converter>` javadoc on this.

Similarly for `@FacesValidator` and `<o:validator>`.

Thanks for reporting!

**Comment by reiudo** [comment]
With OmniFaces 3.1-SNAPSHOT and without the `managed="true"` attribute the converter now works with properties and injected beans. Thanks for fixing!

Is there a release date already planned for OmniFaces 3.1?
